src/main/kotlin/com/openlattice/shuttle/Shuttle.kt

```diff
@@ -463,4 +544,25 @@ class Shuttle(
 
         return if (keyValuesPresent) ApiUtil.generateDefaultEntityId(key.stream(), properties) else ""
     }
+
+    private fun storeLog(flightName: String, log: String, timestamp: OffsetDateTime, status: IntegrationStatus, jobId: UUID) {
+        val logEntitySetId = logEntitySet.id
+        val entityId = logEntitySetId.toString() + flightName + jobId + timestamp.toString()
+        val ek = EntityKey(logEntitySetId, entityId)
+        val logPropertyData = generateLogPropertyData(flightName, log, timestamp, status, jobId)
+        val entity = Entity(ek, logPropertyData)
+        val ekid = idService!!.getEntityKeyId(ek)
+        logsDestination.integrateEntities(setOf(entity), mapOf(ek to ekid), mapOf(logEntitySetId to UpdateType.Merge))
+    }
+
+    private fun generateLogPropertyData(flightName: String, log: String, timestamp: OffsetDateTime, status: IntegrationStatus, jobId: UUID): Map<UUID, Set<Any>> {
```

Comment:
What's the idea behind using `Map<UUID, Set<>>` with `setOf()` instead of `Map<UUID, Any>` without `setOf()` here?

Reply:
I didn't look closely enough at `Entity` to notice the secondary constructor :| will change this

Reply:
Oh wait but then it has to be a SetMultimap

Reply:
Definitely don't use that then. `SetMultimap` is somewhat nuked already and should continue to be hunted down and destroyed

Reply:
What I meant was actually just why use sets if there's only one item in the set ever but I see that it's part of Entity. My bad!